frontend/src/metabase/query_builder/components/view/View/NotebookContainer/NotebookContainer.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from "react";
+
+import Notebook from "metabase/query_builder/components/notebook/Notebook";
+import { Box } from "metabase/ui";
+
+// There must exist some transition time, no matter how short,
+// because we need to trigger the 'onTransitionEnd' in the component
+const delayBeforeNotRenderingNotebook = 10;
+
+interface NotebookContainerProps {
+  isNotebookContainerOpen: boolean;
```

Comment:
While we're at it, we can shorten the name

```suggestion
  isOpen: boolean;
```

Reply:
https://github.com/metabase/metabase/pull/40316/commits/d7546a81cae6d4abc3d57ebab5d0a5df6ec20903